Tests/UnitTests/Plugins/DD4hep/DD4hepCylinderLayerStructureTests.cpp

```diff
@@ -0,0 +1,246 @@
+#include "Acts/Tests/CommonHelpers/CylindricalTrackingGeometry.hpp"
+#include "Acts/Utilities/Logger.hpp"
+
+#include <fstream>
+#include <sstream>
+#include <string>
+#include <tuple>
+
+#include <DD4hep/DetElement.h>
+#include <DD4hep/DetFactoryHelper.h>
+#include <DD4hep/Detector.h>
+#include <XML/Utilities.h>
+#include <XMLFragments.hpp>
+
+#include "DD4hepTestsHelper.hpp"
+
+Acts::GeometryContext tContext;
+Acts::Test::CylindricalTrackingGeometry cGeometry =
+    Acts::Test::CylindricalTrackingGeometry(tContext);
+
+const char* beampipe_head_xml =
+    R""""(
+    <detectors>
+        <detector id="0" name="BeamPipe" type="BarrelDetector">
+            <type_flags type="DetType_TRACKER + DetType_BEAMPIPE"/>
+            <layers>
+                <layer name="BP" id="0">            
+)"""";
+
+const char* cylinder_layer_head_xml =
+    R""""(
+    <detectors>
+        <detector id="1" name="BarrelLayer" type="BarrelDetector" readout="PixelReadout">
+            <type_flags type="DetType_TRACKER + DetType_BARREL"/>
+            <layers>
+                <layer name="B0" id="0">
+)"""";
+
+const char* tail_xml =
+    R""""(
+                </layer>
+            </layers>
+        </detector>
+    </detectors>
+)"""";
+
+const char* indent_12_xml = "            ";
+
+BOOST_AUTO_TEST_SUITE(DD4hepPlugin)
+
+// This tests creates a beampipe as a passive cylinder surface
+BOOST_AUTO_TEST_CASE(DD4hepPluginBeampipeStructure) {
+  // Create an XML from it
+  std::ofstream cxml;
+
+  std::string fNameBase = "BeamPipe";
+
+  cxml.open(fNameBase + ".xml");
+  cxml << head_xml;
+  cxml << beampipe_head_xml;
+  cxml << indent_12_xml << "  <passive_surface>" << '\n';
+  cxml << indent_12_xml
+       << "    <tubs rmin=\"25*mm\" rmax=\"25.8*mm\" dz=\"1800*mm\" "
+          "material=\"Air\"/>"
+       << '\n';
+  cxml << indent_12_xml << "  </passive_surface>" << '\n';
+
+  cxml << tail_xml;
+  cxml << end_xml;
+  cxml.close();
+
+  auto lcdd = &(dd4hep::Detector::getInstance());
+  lcdd->fromCompact(fNameBase + ".xml");
+  lcdd->volumeManager();
+  lcdd->apply("DD4hepVolumeManager", 0, nullptr);
+
+  auto world = lcdd->world();
+
+  // Now the test starts ...
+  auto sFactory = std::make_shared<Acts::DD4hepDetectorSurfaceFactory>(
+      Acts::getDefaultLogger("DD4hepDetectorSurfaceFactory",
+                             Acts::Logging::DEBUG));
+
+  Acts::Experimental::DD4hepLayerStructure beamPipeStructure(
+      std::move(sFactory), Acts::getDefaultLogger("DD4hepBeamPipeStructure",
+                                                  Acts::Logging::VERBOSE));
+
+  Acts::DD4hepDetectorElement::Store dd4hepStore;
+
+  Acts::Experimental::DD4hepLayerStructure::Options lsOptions;
+  lsOptions.name = "BeamPipe";
+  lsOptions.logLevel = Acts::Logging::INFO;
+
+  auto beamPipeInternalsBuilder =
+      beamPipeStructure.builder(dd4hepStore, world, lsOptions);
+
+  // Build the internal volume structure
+  auto [surfaces, volumes, surfacesUpdator, volumeUpdator] =
```

Comment:
Are you not going to run any checks on e.g. the surface, like how many there are etc?

Reply:
Inserted, don't know where they went.